liquibase-standard/src/main/java/liquibase/plugin/AbstractPluginFactory.java

```diff
@@ -65,8 +67,15 @@ protected T getPlugin(final Object... args) {
             }
         }
 
-        if (applicable.size() == 0) {
+        if (applicable.isEmpty()) {
             return null;
+        } else if (applicable.size() == 1) {
+            return applicable.iterator().next();
+        } else if (getPluginClass().equals(Change.class)) {
```

Comment:
I don't like this equals here , but I can't think of something better to address this problem.

Reply:
I think the equals is okay, but it doesn't belong in the plugin factory. It should go in the `ChangeFactory`. Perhaps you need to add a new abstract method called `filter` or something like that, and provide a simple no-op implementation in the plugin factory (for backwards compatibility) and move this logic to the `ChangeFactory` class.

Reply:
Thanks! As I was working with protected methods, I added a new protected one that returns all plugins.